app/routes/events/EventCreateRoute.js

```diff
@@ -1,41 +1,86 @@
-// @flow
-
 import { connect } from 'react-redux';
 import { compose } from 'redux';
 import { formValueSelector } from 'redux-form';
 import { createEvent } from 'app/actions/EventActions';
 import { uploadFile } from 'app/actions/FileActions';
 import EventEditor from './components/EventEditor';
+import {
+  selectEventById,
+  selectPoolsWithRegistrationsForEvent
+} from 'app/reducers/events';
 import { LoginPage } from 'app/components/LoginForm';
 import { transformEvent, time } from './utils';
+import prepare from 'app/utils/prepare';
 import replaceUnlessLoggedIn from 'app/utils/replaceUnlessLoggedIn';
+import { isEmpty } from 'lodash';
+import { fetchEvent } from 'app/actions/EventActions';
+import loadingIndicator from 'app/utils/loadingIndicator';
 
 const mapStateToProps = (state, props) => {
   const actionGrant = state.events.actionGrant;
   const valueSelector = formValueSelector('eventEditor');
+  const eventId = props.location.query.id;
+
+  const eventTemplate = selectEventById(state, { eventId });
+  const poolsTemplate = selectPoolsWithRegistrationsForEvent(state, {
+    eventId
+  });
+
+  const initialValues = {
+    title: '',
+    startTime: time({ hours: 17, minutes: 15 }),
+    endTime: time({ hours: 20, minutes: 15 }),
+    description: '',
+    text: '',
+    eventType: '',
+    company: null,
+    responsibleGroup: null,
+    location: 'TBA',
+    isPriced: false,
+    useStripe: true,
+    priceMember: 0,
+    paymentDueDate: time({ days: 7, hours: 17, minutes: 15 }),
+    mergeTime: time({ hours: 12 }),
+    useCaptcha: true,
+    heedPenalties: true,
+    isAbakomOnly: false,
+    feedbackDescription: 'Melding til arrangører',
+    pools: [],
+    unregistrationDeadline: time({ hours: 12 })
+  };
+
+  if (isEmpty(eventTemplate) && eventId) {
+    return { initialValues: null };
+  }
+
```

Comment:
Maybe rewrite the return logic, to make it a bit more sensible?

```js
if (!eventId) return {initialValues: initialValuesForCreate};
if (isEmpty(eventTemplate)) return {initialValues: null}
return {
    ...stuff
};
```

Reply:
This won't work! You really got me confused with this one 👎Been looking at _Cannot read 'eventType' of undefined for an hour now_.